Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -787,6 +907,24 @@ public class DependencyResolver<
                 guard let constraints = self.safely({ try container.getDependencies(at: version) }) else {
                     return AnySequence([])
                 }
+
+                // Since this is a versioned container, none of its
+                // dependencies can have a revision constraints.
+                let revisionConstraints: [(AnyPackageContainerIdentifier, String)]
+                revisionConstraints = constraints.flatMap{
+                    if case .revision(let revision) = $0.requirement {
+                        return (AnyPackageContainerIdentifier($0.identifier), revision)
+                    }
+                    return nil
+                }
+                // If we have any revision constraints, set the error and abort.
+                guard revisionConstraints.isEmpty else {
+                    self.error = DependencyResolverError.revisionConstraints(
+                        dependency: AnyPackageContainerIdentifier(container.identifier),
+                        revisions: revisionConstraints)
+                    return AnySequence([])
+                }
```

Comment:
This doesn't _feel_ right to me -- I would have expected this to be a natural side effect of other parts of the code. I expect this to return the versions which are *immediately* valid but the filter based on other constraints in the system to happen in response to attempts to merge the binding into the assignment. Does that not work?